### DISCUSSION TITLE
Too large dataset

### DISCUSSION THREAD

**Discussion by theuselesscorporation** [danielegrattarola/spektral#251] [Q&A]
I have a dataset that contains thousands of graphs. I am unable to load them as the Dataset object first wants to load all the graphs.  For batch training how to load data in batches from the disk

**Comment by danielegrattarola** [comment]
Hi, 

Datasets in spektral are supposed to be in-memory, so there is no direct support for what you are asking. 
You can try to return a generator instead of a list from the `read()` function but many methods of the class will likely not work.
